lib/iris/fileformats/pp.py

```diff
@@ -1955,23 +1955,33 @@ def _convert_constraints(constraints):
     constraints = iris._constraints.list_of_constraints(constraints)
     pp_constraints = {}
     unhandled_constraints = False
+
+    def make_func(stashobj):
```

Comment:
I did some digging and it looks like another approach is to pass the `stashobj` variable as a default to a keyword argument for the lambda function:

    call_func = lambda s, stash=stashobj: s==stashobj

I feel like this is quite a confusing way to do it although in any case the situation is quite subtle. It has the benefit of compactness and perhaps would work just as well with a comment explaining what's going on.

Which do you think would be best?